### PR TITLE
Add event based hook to calculate doom (#550)

### PR DESCRIPTION
**Pull request by jmartinez989**
This change is add an event based hook to calcualte doom for any mods
(like LW2) that have missions that generate doom that could potentially
be not revealed causing doom generation updates to no be displayed
properly.

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc [hunk 1]
```diff
@@ -128,12 +128,15 @@ simulated function UpdateMissingPersons()
...
 	local int i;
+	local int CurrentDoom;
+	local int MaxDoom;
 	local XComGameState_HeadquartersAlien AlienHQ;
 	local XComGameState_HeadquartersResistance ResHQ;
 	local XComGameState NewGameState;
 	local bool bPlayedSound;
 	
 	bPlayedSound = false;
+	`log("UpdateData() Start");
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc]: Please no excessive logs

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc [hunk 2]
```diff
@@ -247,7 +254,6 @@ simulated function UpdateData()
 						`XSTRATEGYSOUNDMGR.PlaySoundEvent("Geoscape_DoomDecrease");
 						bPlayedSound = true;
 					}
-
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc]: Please no uncalled-for changes in code style

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_HeadquartersAlien.uc [hunk 3]
```diff
@@ -493,10 +493,27 @@ function int GetCurrentDoom(optional bool bIgnorePending = false)
...
 	local XComGameState_MissionSite MissionState;
 	local int TotalDoom;
+	//#550 - Tuple to send to triggered event.
+	local XComLWTuple Tuple;
 
 	TotalDoom = Doom;
 	History = `XCOMHISTORY;
-	
+
+	//#550 -  set up Tuple for event use.
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_HeadquartersAlien.uc]: Please use the wording "Issue #...", as well as "Begin issue..." and "End issue..."

**Comment by Xymanek** [comment]
Also please don't manually make line breaks in the issue/PR description - let the GitHub website itself handle that

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc (hunk 1 above)
Comment (jmartinez989) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc]: oops. forgot to remove that. will do that. I though I had caught all of those.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc (hunk 2 above)
Comment (jmartinez989) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc]: sorry a bit confused here. could you elaborate? I just want to make sure I understand what is being requested.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_HeadquartersAlien.uc (hunk 3 above)
Comment (jmartinez989) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_HeadquartersAlien.uc]: ok I will make the changes.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc (hunk 2 above)
Comment (Xymanek) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc]: You've removed empty lines from places that your changes do not touch at all - please avoid doing that. It introduces "noise" into your PR, making it harder to review and generally messier

Comment (jmartinez989) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIStrategyMap_HUD.uc]: ahh ok. will keep that in mind. will make the change to correct.

**Comment by jmartinez989** [comment]
Ok I have made the changes. Let me know if there is anything else that might need changing.

**Pull request closed without merging**